nixos/modules/tasks/filesystems/overlayfs.nix

```
@@ -73,10 +76,22 @@ let
         description = ''
           The path to the workdir.
 
+          If the filesystem is `neededForBoot`, this will be prefixed with `/sysroot`,
+          unless `prefixWithSysroot` is set to `false`.
+
           This MUST be set if you set `upperdir`.
         '';
       };
 
+      prefixWithSysroot = lib.mkOption {
```

This sounds a little convoluted but I don't find the current name really explains what it does and it also is not strictly correct (because in the scripted initrd, the path is prefixed with `mnt-root` instead of `sysroot`.

```suggestion
      underlyingDirectoriesOnRealRoot = lib.mkOption {
```